### ISSUE TITLE
Possible breaking change with ReactNative upgrade (0.57 to 0.58) ? 

### ISSUE THREAD

**Issue by MayoudP** [redux-saga/redux-saga#1870]
Hello, 
I just recently upgraded my react-native version from `0.57.7 to 0.58.1`. 
If everything was working well on both my app on iOS and Android on `0.57.7`, just after upgrading, it appears that an API call at the startup of my app, didn't returned anything. 

Weird, because with iOS it's working fine even on the new version, and only on Android, it's not... 

Here is the general syntax : 

```
[...Actions and Reducers involved before...]
export function* watchStartup(api) {
  yield call(doSomething, api);
}

function* doSomething(api) {
{
         const response = yield call(api.doGetOnAPI);
         ==> response not returning anything...
        if (response.ok)
        {...}
}
```

Does anyone reported similar issues when upgrading React-Native ? The API call is a very simple one, a simple get, without ID's or headers needed... 
Still working fine on 0.57.7, but completely stuck with the upgrade to 0.58 

Infos: 
```
    "redux": "4.0.1",
    "redux-logger": "^2.7.4",
    "redux-persist": "^4.6.0",
    "redux-saga": "^0.16.0",
    "redux-thunk": "^2.3.0",
    "reduxsauce": "^0.4.1",
    "react": "16.6.3",
    "react-native": "0.58.1",
```

Thank's for the help 🙏🏻🙏🏻🙏🏻🙏🏻🙏🏻🙏🏻

**Comment by Freddy03h** [comment]
Hi! Experiencing same issue only on Android too.

`take` doesn't catch any actions.

Did you found a workaround ? Or a breaking change we missed ? thank's

```
    "react": "16.8.3",
    "react-native": "0.59.10",

    "react-redux": "7.1.0",
    "redux": "4.0.4",

    "redux-immutable": "4.0.0",
    "redux-localstorage": "1.0.0-rc5",
    "redux-localstorage-debounce": "0.1.0",
    "redux-persist-filesystem-storage": "1.4.0",

    "redux-saga": "1.0.5",

    "reselect": "4.0.0"
```

**Comment by Andarist** [comment]
This really seems like some environmental problem. `redux-saga` doesn't do anything special in node, react-native or in the browser - it's an isomorphic package. I'm afraid we can't help you much, because we don't have the react native expertise - you have to dig into this on your own.

**Comment by Freddy03h** [comment]
@Andarist Thank's for your really quick answer, yes it's an environmental problem. Just trying to found some kind of similar issue on different redux related project.

I found my problem, just in case someone get the same : it was related to `redux-persist-filesystem-storage` I downgraded it to `1.3.2`.

**Comment by Slapbox** [comment]
I'm seeing a lone `yield call(myFunc);` that stops execution indefinitely. No errors.

So I stepped through the functions to try to find the issue, and watched them all complete successfully. But still execution is never resumed after the function is called. So I tried putting `return;` immediately at the top of the problem function - and the behavior is exactly the same. Works everywhere for me except React Native (Android and iOS).

Not really a `redux-saga` problem, but @Andarist if you have any thoughts about debugging this I'd appreciate it greatly. Not sure what to make of this one.

**Comment by Andarist** [comment]
I'm afraid I don't have any immediate tips for debugging this. If `myFunc` is a "sync" generator (no async stuff involved) you should be able to step through redux-saga internals and reach your `log.warn` - I can't imagine how it would not be reached unless there is some bug in redux-saga (rather unlikely, given its stability) or that there is some error thrown there which you miss (for any reason, maybe its suppressed somehow?)

**Comment by Slapbox** [comment]
I figured as much but I've been stuck mightly on this one. React Native's debugging tools leave a lot to be desired.

I did get it working just a minute ago. I cannot understand for the life of me why the simplest possible test case failed. I tried stepping through to the `log.warn` statement, but after a couple hundred steps I gave up. 

No need to read beyond this point, but for posterity in case it helps someone:

Oddly, while the example I gave inexplicably continues to fail in all scenarios (no debugger/browser debugger/[React Native Debugger](https://github.com/jhen0409/react-native-debugger)) - I was able to find the issue in our original code working by inspecting in the browser debugger and changing a sequential `call` & `take` to a `yield all([ call..., take... ])`. No idea why this wasn't apparent to me when I stepped through the code in React Native Debugger.

Since I got the real thing working, I won't question why my test case failed, but wow; React Native development is worse than getting teeth drilled sometimes.
